cmake/compiler_flags.cmake

```diff
@@ -83,7 +83,7 @@ if (MSVC)
 elseif(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   set(CMAKE_CXX_STANDARD 20)
   set(CMAKE_CXX_STANDARD_REQUIRED ON)
-  add_compile_options(-std=c++20)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++20")
```

Comment:
This is not how you require a minimum standard.
This will force a specific standard for every one.
Please use required compile features.

Reply:
Can add but is this actually better? E.g., 20 was not completely backwards compatible with 17.
And a C++23 compiler can also be set to c++20

Reply:
e.g. in add_library_macro we have
  #TODO cxx_std_17 only requires a c++17 flag for the compiler. Not full standard support.
  # If we want full support, we need our own try_compiles (e.g. for structured bindings first available in GCC7)
  # or specify a min version of each compiler.
  target_compile_features(${openms_add_library_TARGET_NAME} PUBLIC cxx_std_17)

Reply:
Yes, if someone uses a compiler whose default is cxx26 you will otherwise force them AND all downstream libraries to use cxx20.

Reply:
And yes, it also removes the need of getting the flags right on different compilers. CMake will do it.